pkg/kubelet/kubelet.go

```diff
@@ -2964,7 +2979,7 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 		}
 		if kl.isPodResizeInProgress(allocatedPod, podStatus) {
 			// If a resize is in progress, make sure the cache has the correct state in case the Kubelet restarted.
-			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
+			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", generationForPodResizeInProgressCondition, false)
```

Comment:
optional: rather than handling `generationForPodResizeInProgressCondition`, you could just check if there is already a `PodResizeInProgress` condition, and skip setting it if there is.

Reply:
Unless I'm misunderstanding your suggestion, I'm not convinced that this is what we want to do. If there is a pod resize in progress, but something else (unrelated to the pod resize) changes in the podspec, the `metadata.generation` will get bumped. In this case, I think the `observedGeneration` of the PodResizeInProgress condition should get bumped too (therefore should not skip setting it). 

My intention is to only preserve the PodResizeInProgress observedGeneration when both conditions are present; otherwise it should reflect the latest `metadata.generation`.